app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,55 @@
+class QueueItemsController < ApplicationController
+
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def update
+
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    if ! current_user_queued_video? video
```

Comment:
You really shouldn't have a space after the `!`, I was surprised to find that works.